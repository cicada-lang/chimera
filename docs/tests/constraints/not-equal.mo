print find _ {
  1 != 1
}

print find _ {
  1 != 2
}

print find [x, y] {
  y != "cat"
  [x, 3] != [5, y]
  [3, x] = [y, 5]
}

print find [x, y] {
  y != "cat"
  [x, 3] != [5, y]
  [x, "cat"] = [5, y]
}

print find [x, y] {
  y != "cat"
  [x, 3] != [5, y]
  [x, 5] = [3, y]
}

print find [x, y] {
  [x, 3] != ["cat", y]
  x = "cat"
}

// disj of not equal

print find [x, y] {
  [x, y] != [1, 2]
}

print find [x, y] {
  [x, y] != [1, 2]
  x = 1
}

print find [x, y] {
  [x, y] != [1, 2]
  x = 1
  y = 2
}

// filter somePairContainsVar

print find x {
  x != 1
  _y != 2
}

// removeSubsumed

print find [x, y] {
  x != 3
  [x, "cat"] != [3, y]
}
