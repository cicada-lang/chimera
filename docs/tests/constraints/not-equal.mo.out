[]
[_.0]
[]
[]
[[3, 5]]
[["cat", _.0] with { NotEqual(_.0, 3) }]
[ 
  [_.0, _.1] with { disj { NotEqual(_.0, 1) NotEqual(_.1, 2) } }
]
[[1, _.0] with { NotEqual(_.0, 2) }]
[]
[_.0 with { NotEqual(_.0, 1) }]
[[_.0, _.1] with { NotEqual(_.0, 3) }]
